starter-code/src/viking.js

```diff
@@ -1,11 +1,95 @@
 // Soldier
-function Soldier() {}
+class Soldier {
+  constructor(healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg;
+  }
+
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+  }
+}
 
 // Viking
-function Viking() {}
+class Viking extends Soldier {
+  constructor(nameViking, healthViking, strengthViking) {
+    super(healthViking, strengthViking);
+    this.name = nameViking;
+    // this.health = healthViking;
+    // this.strength = strengthViking;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return this.name + " has received " + damage + " points of damage";
+    } else {
+      return this.name + " has died in act of combat";
+    }
+  }
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-function Saxon() {}
+class Saxon extends Soldier {
+  constructor(healthSaxon, strengthSaxon) {
+    super(healthSaxon, strengthSaxon);
+    this.health = healthSaxon;
+    this.strength = strengthSaxon;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return "A Saxon has received " + damage + " points of damage";
+    } else {
+      return "A Saxon has died in combat";
+    }
+  }
+}
 
-// War
-function War() {}
+// War;
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    var indexSaxon = Math.floor(Math.random() * this.saxonArmy.length);
```

Comment:
You were really close to being able to pass one of the tests of this iteration 

```
    var indexViking = Math.floor(Math.random() * this.vikingArmy.length);
    var randViking = this.vikingArmy[indexViking];

    var indexSaxon = Math.floor(Math.random() * this.saxonArmy.length);
    var randSaxon = this.saxonArmy[indexSaxon];

    var damage = randViking.attack();
    randSaxon.receiveDamage(damage);

    if (randSaxon.health <= 0) {
      this.saxonArmy.splice(indexSaxon, 1);
      return "A Saxon has died in combat";
    } 
```

Your return value needed to be a little different- it wanted you to return the result of calling your function